frontend/src/pages/leaguePage.js

```diff
@@ -1,77 +1,187 @@
+
 
     const handlePageClick = (event) => {
-        const newOffset = (event.selected * teemsPerPage) % teams.length;
-        setTeamOffset(newOffset);
+        const selectedPage = event.selected;
+        setCurrentPage(selectedPage);
+        getTeams(selectedPage);
+    };
+
+    function handleSearchClick() {
+        setSearchClicked((prev) => !prev);
+    }
+
+    const handleKeyDown = (event) => {
+        if (event.key === 'Enter') {
+            handleSearchClick();
+        }
+    };
+    const getTeams = async (page) => {
+
+        setPrevInputValue(inputValue);
+
+        try {
+            setLoading(true);
+
+            const response = await axios.post(
+                `${apiEndpoints.url}${apiEndpoints.teams.getAll}`,
+                {
+                    teams__sport_id: sportId,
+                    leagues__api_id: leagueId,
+                    letter: inputValue,
+                    page: page + 1,
+                    per_page: teamsPerPage
+                },
+                {
+                    headers: { 'Content-Type': 'application/json' },
+                }
+            );
+
+            setCurrentTeams(response.data.teams);
+            console.log(response.data.teams);
+            const totalPosts = response.data.count;
+            setPageCount(Math.ceil(totalPosts / teamsPerPage));
+        } catch (error) {
+            setPageCount(0);
+            toast.error(`:( Troubles With Leagues Loading: ${error}`);
+        }
     };
 
     useEffect(() => {
-        axios.get(`${apiEndpoints.url}${apiEndpoints.team.getAll}`)
-            .then(res => {
-                let returnedTeams = res.data;
-                res.data.forEach((league) =>{
-                    if(league?.sport === leagueName) {
-                        returnedTeams = league?.team
-                        setTeams(returnedTeams);
-                    }
-                })
-            })
-            .catch(error => {
-                toast.error(`:( Troubles With Teams Loading ${error}`);
-            });
-    }, []);
+        console.log("Sport ID:", sportId);
+        console.log("League ID:", leagueId);
+        if (!sportId || !leagueId) {
+            toast.error("Missing sport or league ID.");
+            navigate("/not-existing");
+            return;
+        }
+        getTeams(0);
+    }, [sportId, leagueName]);
+
+    useEffect(() => {
+        if(prevInputValue !== inputValue){
+            handlePageClick({selected: 0});
+            setPaginationKey((prevKey) => prevKey + 1);
+        }
+    }, [searchClicked]);
+
+    useEffect(() => {
+        (loading === false) ? setLoading(false) :
+            (
+                (currentTeams.length > 0) ? setLoading(false)
+                    : setTimeout(() => {
+                        setLoading(false);
+                    }, 2000)
+            )
+    }, [loading]);
+
+
 
     return(
         <>
+            <section className={"leaguePage"}>
```

Comment:
Leave it as it is now, if you want, 'cause it's still readable (not too much html)

But maybe we could somehow make this html generic? So that we would be able to do something like: customModule(type=teams) /// customModule(type=leagues)  ... you get the idea. It would return the html we need with slight adjustments based on the type we require at the given moment. And then we would be able to reeuse it (mostly at least).

Of course, that is only possible if your leagues/teams/whatever are very similar. If they are not - consider everything written above to be BULLSHIT.

That pagination block just has some "generic" ring to it in my opinion